web/web.go

```diff
@@ -0,0 +1,773 @@
+	if err != nil {
+		return err
+	}
+	if _, err := f.Write(convJSON); err != nil {
+		return err
+	}
+	return nil
+}
+
+func checkPrimaryKeyPrefix(table string, refTable string, fk ddl.Foreignkey, tableInterleaveIssues *TableInterleaveStatus) bool {
+	childPks := app.conv.SpSchema[table].Pks
+	parentPks := app.conv.SpSchema[refTable].Pks
+	if len(childPks) >= len(parentPks) {
+		for i, pk := range parentPks {
+			if i >= len(fk.ReferColumns) || pk.Col != fk.ReferColumns[i] || pk.Col != childPks[i].Col || fk.Columns[i] != fk.ReferColumns[i] {
+				return false
+			}
+		}
+	} else {
+		return false
+	}
+	return true
+}
+
+func remove(slice []string, s int) []string {
+	return append(slice[:s], slice[s+1:]...)
+}
+
+func removePk(slice []ddl.IndexKey, s int) []ddl.IndexKey {
+	return append(slice[:s], slice[s+1:]...)
+}
+
+func removeFk(slice []ddl.Foreignkey, s int) []ddl.Foreignkey {
+	return append(slice[:s], slice[s+1:]...)
+}
+
+func removeColumn(table string, colName string, srcTableName string) {
+	sp := app.conv.SpSchema[table]
+	for i, col := range sp.ColNames {
+		if col == colName {
+			sp.ColNames = remove(sp.ColNames, i)
+			break
+		}
+	}
+	delete(sp.ColDefs, colName)
+	for i, pk := range sp.Pks {
+		if pk.Col == colName {
+			sp.Pks = removePk(sp.Pks, i)
+			break
+		}
+	}
+	srcColName := app.conv.ToSource[table].Cols[colName]
+	delete(app.conv.ToSource[table].Cols, colName)
+	delete(app.conv.ToSpanner[srcTableName].Cols, srcColName)
+	delete(app.conv.Issues[srcTableName], srcColName)
+	app.conv.SpSchema[table] = sp
+}
+
+func renameColumn(newName, table, colName, srcTableName string) {
+	sp := app.conv.SpSchema[table]
+	for i, col := range sp.ColNames {
+		if col == colName {
+			sp.ColNames[i] = newName
+			break
+		}
+	}
+	if _, found := sp.ColDefs[colName]; found {
+		sp.ColDefs[newName] = ddl.ColumnDef{
+			Name:    newName,
+			T:       sp.ColDefs[colName].T,
+			NotNull: sp.ColDefs[colName].NotNull,
+			Comment: sp.ColDefs[colName].Comment,
+		}
+		delete(sp.ColDefs, colName)
+	}
+	for i, pk := range sp.Pks {
+		if pk.Col == colName {
+			sp.Pks[i].Col = newName
+			break
+		}
+	}
+	srcColName := app.conv.ToSource[table].Cols[colName]
+	app.conv.ToSpanner[srcTableName].Cols[srcColName] = newName
+	app.conv.ToSource[table].Cols[newName] = srcColName
+	delete(app.conv.ToSource[table].Cols, colName)
+	app.conv.SpSchema[table] = sp
+}
+
+func pkChanged(pkChange, table, colName string) {
+	sp := app.conv.SpSchema[table]
+	if pkChange == "REMOVED" {
+		for i, pk := range sp.Pks {
+			if pk.Col == colName {
+				sp.Pks = removePk(sp.Pks, i)
+				break
+			}
+		}
+	}
+	if pkChange == "ADDED" {
+		if sPk, found := app.conv.SyntheticPKeys[table]; found {
+			for i, col := range sp.ColNames {
+				if col == sPk.Col {
+					sp.ColNames = remove(sp.ColNames, i)
+					break
+				}
+			}
+			delete(sp.ColDefs, sPk.Col)
+			for i, pk := range sp.Pks {
+				if pk.Col == sPk.Col {
+					sp.Pks = removePk(sp.Pks, i)
+					break
+				}
+			}
```

Comment:
Shouldn't we also delete the entry from the SyntheticPKeys map?

Maybe something like:
```
  delete(app.conv.SyntheticPKeys, table)
```

Reply:
yes, thats required, thanks